assimilation_code/modules/assimilation/algorithm_info_mod.f90

```diff
@@ -0,0 +1,620 @@
+! DART software - Copyright UCAR. This open source software is provided
+! by UCAR, "as is", without charge, subject to all terms of use at
+! http://www.image.ucar.edu/DAReS/DART/DART_download
+
+module algorithm_info_mod
+
+! Provides routines that give information about details of algorithms for 
+! observation error sampling, observation increments, and the transformations
+! for regression and inflation in probit space. 
+
+use types_mod,       only : r8, i8, MISSING_R8, obstypelength
+
+use obs_def_mod,     only : obs_def_type, get_obs_def_type_of_obs, get_obs_def_error_variance
+use obs_kind_mod,    only : get_quantity_for_type_of_obs, get_name_for_quantity, get_index_for_quantity
+
+use utilities_mod,   only : error_handler, E_ERR, E_MSG, open_file, close_file, to_upper, &
+                            do_nml_file, do_nml_term, nmlfileunit, check_namelist_read, &
+                            find_namelist_in_file
+
+use assim_model_mod, only : get_state_meta_data
+use location_mod,    only : location_type
+
+use distribution_params_mod, only : NORMAL_DISTRIBUTION, BOUNDED_NORMAL_RH_DISTRIBUTION, &
+   GAMMA_DISTRIBUTION, BETA_DISTRIBUTION, LOG_NORMAL_DISTRIBUTION, UNIFORM_DISTRIBUTION,  &
+   PARTICLE_FILTER_DISTRIBUTION
+
+implicit none
+private
+
+character(len=512) :: errstring
+character(len=*), parameter :: source = 'algorithm_info_mod.f90'
+
+logical :: module_initialized = .false.
+logical :: use_qty_defaults = .true.
+
+! Defining parameters for different observation space filters
+integer, parameter :: EAKF               = 1
+integer, parameter :: ENKF               = 2
+integer, parameter :: KERNEL             = 3
+integer, parameter :: OBS_PARTICLE       = 4
```

Comment:
KERNEL, OBS_PARTICLE are not selectable as filter_kinds
see line 282 case statement 
      case ('EAKF')
      case ('ENKF')
      case ('UNBOUNDED_RHF')
      case ('GAMMA_FILTER')
      case ('BOUNDED_NORMAL_RHF')